src/impl/realm_coordinator.cpp

```diff
@@ -348,9 +351,9 @@ void RealmCoordinator::commit_write(Realm& realm)
         m_notifier->notify_others();
     }
 #if REALM_ENABLE_SYNC
-    if (m_sync_session) {
+    if (m_sync_session && !realm.is_closed()) {
```

Comment:
Shouldn't this be moved to before the notifications? I wouldn't expect calling `close()` on a Realm to stop it from syncing the commit I just made.